src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -147,6 +149,16 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
 
     const [isBackendKilled, setIsBackendKilled] = useState(false);
 
+    const [outputDataMap, setOutputDataMap] = useState<Map<string, OutputData>>(
+        new Map<string, OutputData>()
+    );
```

Comment:
Let's go full immutable. `EMPTY_MAP` is from util.

```suggestion
    const [outputDataMap, setOutputDataMap] = useState<ReadonlyMap<string, OutputData>>(EMPTY_MAP);
```